internal/templates/hook.tmpl

```diff
@@ -15,7 +19,10 @@ call_lefthook()
   osArch=$(uname | tr '[:upper:]' '[:lower:]')
   cpuArch=$(uname -m | sed 's/aarch64/arm64/')
 
-  if lefthook{{.Extension}} -h >/dev/null 2>&1
+  if test -n "$LEFTHOOK_PATH"
+  then
+    "$LEFTHOOK_PATH" "$@"
```

Comment:
Hey! Thank you for this PR, it makes sense to add this feature. I think it's not obvious that LEFTHOOK_PATH is not a path to a folder containing `lefthook` executable, but a full path to the lefthook executable. 

I wonder if there's a better name for this feature 🤔 WDYT about `LEFTHOOK_BIN` or `LEFTHOOK_EXECUTABLE`?

Reply:
I went with `_BIN`, on my very scientific survey (looking at neovide, which I happened to have open)